New message indicator

gabriel opened the pull request:
Adds a new message indicator.

![2016-11-18 at 4 09 pm](https://cloud.githubusercontent.com/assets/2669/20450963/7e22dd6a-ada9-11e6-8511-5f4b6bd19492.png)

- When messages are appended to a conversation, it marks the first new message.
- When new messages are appended:
  - If conversation not visible, if new message is not set, set it  
  - If conversation is visible, clear new message (if one is set)
- When you select a different conversation it clears the first new message marker of the conversation you were just on (you've seen the messages). I take care to only do this when a user selects a conversation through click action (if the conversation select was programmatic it won't clear the indicator).

gabriel commented:
@keybase/react-hackers

chrisnojima commented:
Can we add a TODO that the logic for the currently selected message. If you get a message on the currently selected conversation and the electron window isn't focused then we actually do get the orange line. This isn't MVP but is something we'll want later.

chrisnojima commented on shared/chat/conversation/messages/text.desktop.js:
```diff
@@ -56,4 +56,8 @@ const MessageTextWrapper = ({author, message, messageState, followState, style,
 )
 
+const stylesFirstNewMessage = {
+  borderTop: 'solid 1px #ff6f21',
```
```
`solid 1px ${globalColors.orange}`
```

chrisnojima commented:
Doesn't seem like the currently selected logic is working for me. If i have a conversation selected and i get a new message it draws the orange line anyways

keybase-ci-visdiff commented:
The commits 375194da062c...e034f82611b5 introduce visual changes on linux. <details><summary>:mag_right: 3 changed</summary>

 * Changed: **unlock_folders-Single** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-Single-375194da062c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-Single-e034f82611b5.png)  
   ![unlock_folders-Single comparison](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-Single.png)
 * Changed: **unlock_folders-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-Normal-375194da062c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-Normal-e034f82611b5.png)  
   ![unlock_folders-Normal comparison](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-Normal.png)
 * Changed: **unlock_folders-No_paperkeys** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-No_paperkeys-375194da062c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-No_paperkeys-e034f82611b5.png)  
   ![unlock_folders-No_paperkeys comparison](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480360953997-375194da062c-e034f82611b5-linux/unlock_folders-No_paperkeys.png)
</summary>

gabriel commented:
Ok fixed the issue where new message wasn't clearing indicator... I updated the PR description rules to reflex this. Also added some more comments in code.
